python/az/aro/azext_aro/custom.py

```diff
@@ -138,43 +150,51 @@ def aro_create(cmd,  # pylint: disable=too-many-locals
 
 def aro_delete(cmd, client, resource_group_name, resource_name, no_wait=False):
     # TODO: clean up rbac
-
     try:
         oc = client.get(resource_group_name, resource_name)
 
-        master_subnet = oc.master_profile.subnet_id
-        worker_subnets = {w.subnet_id for w in oc.worker_profiles}
-
-        master_parts = parse_resource_id(master_subnet)
-        vnet = resource_id(
-            subscription=master_parts['subscription'],
-            resource_group=master_parts['resource_group'],
-            namespace='Microsoft.Network',
-            type='virtualNetworks',
-            name=master_parts['name'],
-        )
+        # Get associated cluster resources we need to assign network contributor on
+        resources = get_cluster_network_resources(cmd.cli_ctx, oc)
 
         aad = AADManager(cmd.cli_ctx)
         rp_client_id = os.environ.get('AZURE_FP_CLIENT_ID', FP_CLIENT_ID)
-        rp_client_sp = aad.get_service_principal(rp_client_id)
+        rp_client_sp = None
```

Comment:
surely `rp_client_sp = None` is necessary, but actually it needs to be right at the top of the function before the first try: ?
Otherwise if client.get fails, rp_client_sp is not set and then when we try to read it below we'll break?